src/com/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -911,7 +1014,11 @@
    24 add-producer-timestamps
    25 structured-facts
    26 structured-facts-deferrable-constraints
-   27 switch-value-string-index-to-gin})
+   27 switch-value-string-index-to-gin
+   28 #(let [time-msg (with-out-str
+                        (time (apply lift-fact-paths-into-facts %&)))]
+         (log/info (str "lift: " time-msg)))
```

Comment:
I'd say we strip it out here and implement in on every migration in master. Definitely a cool touch.